supplementary/envelopes_analysis/field_envelopes.py

```diff
@@ -24,26 +24,23 @@
 
 # plotting different envelopes of electric field and intensity, and their spectra
 print("\nEnvelope profiles and spectra calculated")
-colors = plt.cm.viridis(np.linspace(0, 0.9, len(envelope_types)))
+colormap = plt.cm.viridis(np.linspace(0, 0.9, len(envelope_types)))
+colors = dict(zip(envelope_types, colormap))
 fig, axs = plt.subplots(2, 2, figsize=(7, 7))
 
-for i in range(len(envelope_types)):
-    envelope_type = envelope_types[i]
+for envelope_type in envelope_types:
     print(f"* {envelope_type}")
 
-    if not envelope_type in ENVELOPE_TYPES:
-        print(f"ERROR: envelope type '{envelope_type}' not available in promdens.py")
-        exit(1)
-
     pulse = LaserPulse(omega=omega, fwhm=fwhm, t0=t0, lchirp=lchirp, envelope_type=envelope_type)
     field = InitialConditions()
     field.calc_field(pulse=pulse)
     field.field_ft_omega -= omega
 
-    axs[0, 0].plot(field.field_t/field.fstoau, field.field_envelope, color=colors[i], label=f"'{envelope_type}'")
-    axs[0, 1].plot(field.field_t/field.fstoau, field.field_envelope**2, color=colors[i], label=f"'{envelope_type}'")
-    axs[1, 0].plot(field.field_ft_omega/field.evtoau, field.field_ft, color=colors[i], label=f"'{envelope_type}'")
-    axs[1, 1].plot(field.field_ft_omega/field.evtoau, field.field_ft**2, color=colors[i], label=f"'{envelope_type}'")
+    color = colors[envelope_type]
+    axs[0, 0].plot(field.field_t/field.fstoau, field.field_envelope, color=color, label=envelope_type)
+    axs[0, 1].plot(field.field_t/field.fstoau, field.field_envelope**2, color=color, label=envelope_type)
+    axs[1, 0].plot(field.field_ft_omega/field.evtoau, field.field_ft, color=color, label=envelope_type)
+    axs[1, 1].plot(field.field_ft_omega/field.evtoau, field.field_ft**2, color=color, label=envelope_type)
```

Comment:
I see you changed from `label=f"'{envelope_type}'"` to `label=envelope_type`. Could you do the same for the Wigner plot? I mean line 115 `axs[0, i].set_title(f"'{envelope_type}'")`. Then we are consistent.